src/Microsoft.SqlTools.ServiceLayer/Connection/Contracts/SerializeConnectionStringRequest.cs

```diff
@@ -0,0 +1,19 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using Microsoft.SqlTools.Hosting.Protocol.Contracts;
+
+namespace Microsoft.SqlTools.ServiceLayer.Connection.Contracts
+{
+    /// <summary>
+    /// Serialize Connection String request 
+    /// </summary>
+    public class SerializeConnectionStringRequest
+    {
+        public static readonly
+            RequestType<string, ConnectionDetails> Type =
+            RequestType<string, ConnectionDetails>.Create("connection/serializeconnectionstring");

```

Comment:
I'm going back-and-forth whether `serialize` is really the correct verb here (verses something like parse).  I think it's fine, since we are "serializing" into an object, though I typically think of creating objects from a raw value as "deserialize".  In any case, just food for thought.

Reply:
Maybe `buildConnectionInfo` might be more telling?

Reply:
@anthonydresser that sounds good as well, though the convention has been to use all lower-case for JSON-RPC method names (e.g., `"connection/buildconnectioninfo"`), though I realize that isn't followed consistently across all feature areas.